internal/handlers/tigris/tigrisdb/query.go

```diff
@@ -95,25 +110,45 @@ func (tdb *TigrisDB) BuildFilter(filter *types.Document) driver.Filter {
 	res := map[string]any{}
 
 	for k, v := range filter.Map() {
-		// filter only by _id for now
-		if k != "_id" {
+		// don't pushdown empty strings yet
+		if v == "" {
 			continue
 		}
 
-		switch v.(type) {
-		case string:
-			// filtering by string values is complicated if the storage supports encodings, collations, etc,
-			// but Tigris does not support any of these
-		case types.ObjectID:
-			// filtering by ObjectID is always safe
-		default:
-			// skip other types for now
+		if k != "" {
+			// don't pushdown $comment and dot notation yet
+			if k[0] == '$' || types.NewPathFromString(k).Len() > 1 {
+				continue
+			}
+		}
+
+		// _id field supports only specific types
+		if k == "_id" {
+			switch v.(type) {
+			case *types.Document, *types.Array, types.Binary, bool, time.Time, types.NullType, types.Regex, types.Timestamp:
+			// type not supported for pushdown
+
```

Comment:
Formatting in this switch-case looks a bit insocnistent to me: 
- Should each `case` be ended with an empty line or not?
- Should a comment be shifter in this case (like in the next one)?

In the switch below, the same things are formated differently.

Reply:
> Should each case be ended with an empty line or not?

I think if we agree we can put that logic into our linter :hugs: